state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2162,116 +2440,96 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
-				deposit_amount.into(),
-				source_asset.into(),
-				destination_asset,
-			) {
-				Ok(ccm_swap_metadata) => SwapRequestType::Ccm {
-					ccm_swap_metadata,
-					output_address: destination_address_internal.clone(),
-				},
-				Err(reason) => {
-					ccm_failed(reason);
-					return;
-				},
-			}
-		} else {
-			SwapRequestType::Regular { output_address: destination_address_internal.clone() }
-		};
+		}
 
 		if let Err(err) =
 			T::SwapLimitsProvider::validate_refund_params(refund_params.retry_duration)
 		{
-			log::warn!(
-				"Failed to process vault swap due to invalid refund params. Tx id: {tx_id:?}. Error: {err:?}",
-			);
+			log::warn!("Failed to process vault swap due to invalid refund params. Tx id: {tx_id:?}. Error: {err:?}");
 			return;
 		}
 
 		if let Some(params) = &dca_params {
 			if let Err(err) = T::SwapLimitsProvider::validate_dca_params(params) {
-				log::warn!(
-    				"Failed to process vault swap due to invalid dca params. Tx id: {tx_id:?}. Error: {err:?}",
-    			);
+				log::warn!("Failed to process vault swap due to invalid dca params. Tx id: {tx_id:?}. Error: {err:?}");
 				return;
 			}
 		}
 
-		let primary_broker = broker_fee.account.clone();
-
-		let broker_fees =
-			if T::AccountRoleRegistry::has_account_role(&primary_broker, AccountRole::Broker) {
-				[broker_fee]
-				.into_iter()
-				.chain(affiliate_fees.into_iter().filter_map(
-					|Beneficiary { account: short_affiliate_id, bps }| {
-						if let Some(affiliate_id) =
-							T::AffiliateRegistry::lookup(&primary_broker, short_affiliate_id)
-						{
-							Some(Beneficiary { account: affiliate_id, bps })
-						} else {
-							// In case the entry not found, we ignore the entry, but process the
-							// swap (to avoid having to refund it).
-							Self::deposit_event(Event::<T, I>::UnknownAffiliate {
-								broker_id: primary_broker.clone(),
-								short_affiliate_id,
-							});
-
-							None
-						}
-					},
-				))
-				.collect::<Vec<_>>()
-				.try_into()
-				.expect("must fit since affiliates are limited to 1 fewer element than beneficiaries")
-			} else {
-				Self::deposit_event(Event::<T, I>::UnknownBroker { broker_id: primary_broker });
-				Default::default()
-			};
+		let broker = broker_fee.account.clone();
+		let broker_fees = Self::assemble_broker_fees(broker_fee, affiliate_fees);
 
 		if let Err(err) = T::SwapLimitsProvider::validate_broker_fees(&broker_fees) {
-			log::warn!(
-				"Failed to process vault swap due to invalid broker fees. Tx id: {tx_id:?}. Error: {err:?}",
- 			);
+			log::warn!("Failed to process vault swap due to invalid broker fees. Tx id: {tx_id:?}. Error: {err:?}");
 			return;
 		}
 
-		T::SwapRequestHandler::init_swap_request(
-			source_asset.into(),
-			deposit_amount.into(),
-			destination_asset,
-			request_type,
-			broker_fees,
-			Some(refund_params),
-			dca_params,
-			swap_origin,
-		);
+		let (action, source_address) = if let Some(deposit_metadata) = deposit_metadata {
```

Comment:
I think a cleaner solution for extracting/passing `source_address` might be to create a different struct e.g. `DepositAction` which will include `CcmDepositMetadata` instead of `CcmChannelMetadata` in `ChannelAction` (and/or making ChannelAction generic over metadata, though this would make the code less readable), but I chose to go with the solution that requires fewer code for now.